parsec/basicclient.go

```diff
@@ -110,7 +123,7 @@ func (c *BasicClient) selectDefaultProvider() error {
 
 // selectDefaultAuthenticator will request the list of authenticators from the parsec service and will then select
 // the first one it is able to configure automatically
-func (c *BasicClient) selectDefaultAuthenticator() error {
+func (c *BasicClient) selectDefaultAuthenticator(config *ClientConfig) error {
```

Comment:
Maybe `config` and what you store inside `ClientConfig.authenticatorData` could be just strings? Currently the direct authentication is the only method which requires the user to input what's their application name. The other authentication mechanisms just fetch it independently.

Reply:
the reason for that, was to allow for future authenticators - if they need more complex data types, then this code can support this without changing the interface (only adding to it)

Reply:
Ok, I'm fine with it!